CMakeLists.txt

```diff
@@ -163,6 +163,55 @@ option(BUILD_TESTING "Whether to enable tests" ON)
 
 option(BUILD_W3CTRACECONTEXT_TEST "Whether to build w3c trace context" OFF)
 
+option(MAINTAINER_MODE "Build in maintainer mode (-Wall -Werror)" OFF)
+
+if(MAINTAINER_MODE)
+  if(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+    add_compile_options(-Wall)
+    add_compile_options(-Werror)
+    add_compile_options(-Wextra)
+
+    # Tested with GCC 9.4 on github.
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 9.4)
+      # Relaxed warnings
+
+      # Enforced warnings
+    endif()
+
+  endif()
+
+  if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Comment:
Use `elseif` is probably more compact here?

Reply:
Fixed.